portfolio/src/main/java/com/google/sps/servlets/NumCommentsServlet.java

```diff
@@ -54,24 +56,23 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    List<Integer> numComments = new ArrayList<> ();
+    Map<Integer, Integer> numComments = new HashMap<Integer, Integer> ();
 
     // Receive input from the pagination to see which comments to show
     FetchOptions entitiesLimit = FetchOptions.Builder.withLimit(COMMENT_LIMIT);
     double totalComments = results.countEntities(entitiesLimit);
 
     if (totalComments > 0) {
-      int countSize = 0;
+      int currentNumber = 0;
```

Comment:
I don't think you need `currentNumber` here. Instead of using this as a lookback mechanism in the loop, you can do something like this:

```
for (Entity comment : commentsList) {
  if (numComments.containsKey(blogNumber)) {
    numComments.put(blogNumber, numComments.get(blogNumber) + 1);
  } else {
    numComments.put(blogNumber, 1);
  }
}
```